packages/@react-aria/combobox/src/useComboBox.ts

```diff
@@ -157,8 +157,8 @@ export function useComboBox<T>(props: AriaComboBoxOptions<T>, state: ComboBoxSta
   };
 
   let onBlur = (e: FocusEvent<HTMLInputElement>) => {
-    // Ignore blur if focused moved to the button or into the popover.
-    if (e.relatedTarget === buttonRef?.current || popoverRef.current?.contains(e.relatedTarget)) {
+    // Ignore blur if focused moved to the button(if exists) or into the popover.
+    if ((buttonRef?.current && e.relatedTarget === buttonRef?.current) || popoverRef.current?.contains(e.relatedTarget)) {
```

Comment:
```suggestion
    let blurFromButton = buttonRef.current && buttonRef.current === e.relatedTarget;
    let blurIntoPopover = popoverRef.current?.contains(e.relatedTarget);
    // Ignore blur if focused moved to the button(if exists) or into the popover.
    if (blurFromButton || blurIntoPopover) {
      return;
    }
```

Splitting the logic into separate checks could make things a bit clearer to read!